File: src/desktop-edge-agent/src/FccDesktopAgent.Core/MasterData/Models/LocalPump.cs
namespace FccDesktopAgent.Core.MasterData.Models;

/// <summary>
/// Locally cached pump record derived from nozzle mappings in the cloud config.
/// Stores the Odoo-to-FCC pump number mapping.
/// </summary>
public sealed class LocalPump
{
    public int OdooPumpNumber { get; set; }
    public int FccPumpNumber { get; set; }
}

File: src/edge-agent/app/src/main/kotlin/com/fccmiddleware/edge/buffer/entity/LocalPump.kt
package com.fccmiddleware.edge.buffer.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Locally cached pump record derived from nozzle mappings in the cloud config.
 *
 * Stores the Odoo ↔ FCC pump number mapping and an optional display name.
 * Replaced in full on each config refresh.
 */
@Entity(tableName = "local_pumps")
data class LocalPump(
    @PrimaryKey
    @ColumnInfo(name = "odoo_pump_number")
    val odooPumpNumber: Int,

    @ColumnInfo(name = "fcc_pump_number")
    val fccPumpNumber: Int,

    @ColumnInfo(name = "display_name")
    val displayName: String = "",
)
